AutomatticTracks/src/main/java/com/automattic/android/tracks/CrashLogging/BeforeSendCallback.kt

```diff
@@ -0,0 +1,17 @@
+package com.automattic.android.tracks.CrashLogging
+
+import io.sentry.SentryEvent
+import io.sentry.SentryOptions
+
+class BeforeSendCallback(
+        private val debug: Boolean,
+        private val dataProvider: CrashLoggingDataProvider,
+) : SentryOptions.BeforeSendCallback {
+    override fun execute(event: SentryEvent, hint: Any?): SentryEvent? {
+        return if (debug || dataProvider.userHasOptedOut) {
+            null
+        } else {
+            event
+        }
+    }
+}
```

Comment:
I think it'd be better to expose a callback to the clients so they can be the ones who decide if a Sentry event should be sent. We also do [a lot of configuration in the WPAndroid in this callback](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/util/CrashLogging.kt#L32-L82), which I don't think this interface currently allows to do. Actually, can we please make this an interface if we end up keeping it?

I _think_ we may have to expose the SentryEvent, at least partially to the client to be able to do all that configuration. I haven't given this any thought yet, but let me know if you need any help with it.

P.S: I think maybe this is out of the scope for this PR and you may be planning to make these types of changes in a future PR. If that's the case, you can ignore it for this PR, but maybe it'll be helpful for the future one.

Reply:
This class was meant to have `internal` access modifier - I didn't plan to expose it to client, I've extracted it mostly to split responsibilities and unit test it.

> I think we may have to expose the SentryEvent, at least partially to the client to be able to do all that configuration.

That's probably true, your link to WPAndroid illustrates it perfectly but maybe we could abstract it instead of coupling `WPAndroid` with `Sentry` - I'll think of it in future PR.

> but maybe it'll be helpful for the future one.

Sure it is, thanks!

Reply:
OK, it's a bit difficult to give any meaningful feedback on this before we get a sense of what you have in mind, so I'll wait to follow up on this.

Reply:
I've ended up with not extracting this callback for now. I'll follow up to your comments in next PRs.